src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceContextHost.WorkspaceContextHostInstance.cs

```diff
@@ -0,0 +1,111 @@
+                                                IProjectThreadingService threadingService,
+                                                IUnconfiguredProjectTasksService tasksService,
+                                                IProjectSubscriptionService projectSubscriptionService,
+                                                Lazy<IWorkspaceProjectContextProvider> workspaceProjectContextProvider,
+                                                ExportFactory<IApplyChangesToWorkspaceContext> applyChangesToWorkspaceContextFactory)
+                : base(threadingService.JoinableTaskContext)
+            {
+                _project = project;
+                _projectSubscriptionService = projectSubscriptionService;
+                _threadingService = threadingService;
+                _tasksService = tasksService;
+                _workspaceProjectContextProvider = workspaceProjectContextProvider;
+                _applyChangesToWorkspaceContextFactory = applyChangesToWorkspaceContextFactory;
+            }
+
+            public Task InitializeAsync()
+            {
+                return InitializeAsync(CancellationToken.None);
+            }
+
+            protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+            {
+                _context = await _workspaceProjectContextProvider.Value.CreateProjectContextAsync(_project)
+                                                                       .ConfigureAwait(true);
+
+                if (_context == null)
+                    return;
+
+                _applyChangesToWorkspaceContext = _applyChangesToWorkspaceContextFactory.CreateExport();
+                _applyChangesToWorkspaceContext.Value.Initialize(_context);
+
+                _subscriptions = new DisposableBag(CancellationToken.None);
+                _subscriptions.AddDisposable(_projectSubscriptionService.ProjectRuleSource.SourceBlock.LinkToAsyncAction(
+                        target: e => OnProjectChangedAsync(e, evaluation: true),
+                        ruleNames: _applyChangesToWorkspaceContext.Value.GetEvaluationRules()));
+
+                _subscriptions.AddDisposable(_projectSubscriptionService.ProjectBuildRuleSource.SourceBlock.LinkToAsyncAction(
+                        target: e => OnProjectChangedAsync(e, evaluation: false),
+                        ruleNames: _applyChangesToWorkspaceContext.Value.GetDesignTimeRules()));
+            }
+
+            protected override async Task DisposeCoreUnderLockAsync(bool initialized)
+            {
+                if (initialized)
+                {
+                    _subscriptions?.Dispose();
+                    _applyChangesToWorkspaceContext?.Dispose();
+
+                    if (_context != null)
+                    {
+                        await _workspaceProjectContextProvider.Value.ReleaseProjectContextAsync(_context)
+                                                                    .ConfigureAwait(true);
```

Comment:
unnecessary

Reply:
Necessary until it meets the change in master - breaks the build.